### PR TITLE
fix the running exception on pf4j-spring demo #71

### PR DESCRIPTION
**Pull request by liulei3**
1 add the maven dependency javax.annotation javax.annotation-api;
2 add the reference "java.util.List" in org.pf4j.spring.SpringPluginManager

### CODE REVIEW

File: pf4j-spring/src/main/java/org/pf4j/spring/SpringPluginManager.java [hunk 1]
```diff
@@ -24,6 +24,7 @@
 
 import javax.annotation.PostConstruct;
 import java.nio.file.Path;
+import java.util.List;
```
Comment (decebals) [inline pf4j-spring/src/main/java/org/pf4j/spring/SpringPluginManager.java:27]: Good catch

File: pf4j-spring/pom.xml [hunk 2]
```diff
@@ -33,6 +33,12 @@
             <version>${spring.version}</version>
         </dependency>
 
+        <dependency>
```
Comment (decebals) [inline pf4j-spring/pom.xml:36]: This library is built by default with Java 8.
I suggest you to put this dependency as runtime dependency,  in your application.
Other variant is to use profile (Java 9+) dependency (for details see [this](https://stackoverflow.com/a/167284))

**Comment by liulei3** [comment]
update: add the jdk9-build profile declaration the dependency of javax.annotation inside pom.xml[https://github.com/pf4j/pf4j-spring/blob/9f1b4492044dd321317c594053cc38f5032e4f04/pom.xml](url)

**Comment by decebals** [comment]
It looks good. Have you tested for java9 +, is it OK?

**Comment by liulei3** [comment]
Yes,I tested on the jdk version that is 11.0.13 .



***@***.***
 
From: Decebal Suiu
Date: 2022-03-03 16:31
To: pf4j/pf4j-spring
CC: liulei3; Author
Subject: Re: [pf4j/pf4j-spring] fix the running exception on pf4j-spring demo #71 (PR #72)
It looks good. Have you tested for java9 +, is it OK?
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by decebals** [comment]
Thanks!

**Pull request merged**